tests/DCPS/MultiDPTest/publisher.cpp

```diff
@@ -258,23 +239,6 @@ int ACE_TMAIN(int argc, ACE_TCHAR *argv[])
       ACE_OS::fprintf(writers_completed, "%d\n", timeout_writes);
     }
 
-    // Wait for the subscriber to finish.
-    FILE* readers_completed = 0;
-    do {
-      ACE_OS::sleep(small_time);
-      readers_completed = ACE_OS::fopen(sub_finished_filename.c_str(), ACE_TEXT("r"));
-    } while (!readers_completed);
-
-    if (writers_completed) ACE_OS::fclose(writers_completed);
```

Comment:
We should keep this line so that the file is closed before the publisher ends.

Reply:
@mitza-oci In the case of RELIABLE_RELIABILITY_QOS, if the publisher exits before all of its samples have been received by the subscriber, will the rest of the samples still be received by the subscriber eventually?

Reply:
Yes. Thanks for catching that.

Reply:
> @mitza-oci In the case of RELIABLE_RELIABILITY_QOS, if the publisher exits before all of its samples have been received by the subscriber, will the rest of the samples still be received by the subscriber eventually?

Samples that are already in the reader's cache are still available, but the reader's QoS settings will influence this too.  Samples that have not been transferred to the reader's cache do not outlive the writer unless you use a higher level of Durability (which would not be supported in a Safety Profile build).

Reply:
Thanks. So I think there is an issue with these changes. The publisher can exit before the subscriber receives all samples, and if that happens it'll cause the subscriber to timeout. I think we should wait for the subscriber to finish here, for example, using `WaitSet` since we are getting rid of `sub_finished_filename` file that was used to indicate the subscriber's completion. That's the last major comment I have.